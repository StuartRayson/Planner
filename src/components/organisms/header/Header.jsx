import classNames from 'classnames';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Heading from '../../atoms/heading/Heading.jsx';
import Image from '../../atoms/image/Image.jsx';
import Button from '../../atoms/button/Button.jsx';

import logo from '../../../shared/assets/images/logo.svg';
import './header.scss';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: !this.props.closed
    };
    this.handleToggle = this.handleToggle.bind(this);
  }

  handleToggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  render() {
    const classes = classNames(
      'o-header',
      !this.state.isOpen && 'o-header--closed',
      this.props.className
    );

    return (
      <header className={classes}>
        <div className="o-header__content">
          <Image
            className="o-header__logo"
            src={logo}
            height={80}
            alt="React logo"
          />
          <Heading
            className="o-header__title"
            text="Welcome to Inviqa React Starter"
          />
        </div>
        <Button
          className="o-header__toggle"
          onClick={this.handleToggle}
          variant="primary"
        >
        toggle <span role="img" aria-label="smiling face with sunglasses">ðŸ˜Ž</span>
        </Button>
      </header>
    );
  }
}

Header.propTypes = {
  className: PropTypes.string,
  closed: PropTypes.bool
};

export default Header;
