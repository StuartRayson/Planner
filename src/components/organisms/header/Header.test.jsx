import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header.jsx';

describe('<Header />', () => {
  it('renders without crashing', () => {
    shallow(<Header />);
  });

  it('menu should toggle', () => {
    const header = shallow(<Header />);
    header.setState({ isOpen: true });
    header.instance().handleToggle();
    expect(header.instance().state.isOpen).toEqual(false);
  });

  it('should render and match snapshot with basic config', () => {
    const wrapper = shallow(
      <Header />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
