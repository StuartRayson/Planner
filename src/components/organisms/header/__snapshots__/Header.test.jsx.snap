// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match snapshot with basic config 1`] = `
<header
  className="o-header"
>
  <div
    className="o-header__content"
  >
    <Image
      alt="React logo"
      className="o-header__logo"
      height={80}
      src="logo.svg"
    />
    <Heading
      className="o-header__title"
      text="Welcome to Inviqa React Starter"
    />
  </div>
  <Button
    className="o-header__toggle"
    onClick={[Function]}
    variant="primary"
  >
    toggle 
    <span
      aria-label="smiling face with sunglasses"
      role="img"
    >
      ðŸ˜Ž
    </span>
  </Button>
</header>
`;
